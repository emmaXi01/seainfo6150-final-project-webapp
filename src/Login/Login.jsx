import React, {useState} from 'react';
import PropTypes from "prop-types";
import login from "../images/login.png";
import {checkUser} from "../service.js";
import styles from "./Login.module.css";

const Login = (props) => {
  const[username, setUsername] = useState('');
  const[password, setPassword] = useState('');
  const [errorText, setErrorText] = useState('');

  const onUsernameChange = (e) => {
      setUsername(e.target.value);
  };

  const onPasswordChange = (e) => {
    setPassword(e.target.value);
  };

  const performLogin = () => {
    if(!username || username.trim() === "") {
        setErrorText("You should Enter an User Name!");
        return;
    }

    if(!password || password.trim() === "") {
      setErrorText("You should Enter a Password!");
      return;
    }

    if(!checkUser(username, password)) {
      setErrorText("user name or password is wrong!");
      return;
    }

    setErrorText('');
    props.onLogin(username);
  };

  return(
    <div className={styles.loginContainer}>
      <div className={styles.picture}>
          <img className={styles.loginImage} src={login} alt="login"/>
      </div>
      <h4 className={styles.greeting}>hello :)</h4>
      <p className="error">{errorText}</p>
      <div className="login-body">
          <input className={styles.username} onChange={onUsernameChange} value={username} placeholder="User name"/>
          <input className={styles.password} onChange={onPasswordChange} value={password} placeholder="Password"/>
          <button className={styles.loginButton} onClick={ performLogin }>Log In</button>    
      </div>       
    </div>
  );
}

Login.propTypes = {
  onLogin: PropTypes.func.isRequired
};

export default Login;