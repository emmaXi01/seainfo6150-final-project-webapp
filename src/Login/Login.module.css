.loginContainer {
  width: 500px;
  margin: 5% auto 5%;
  padding: 50px 80px;
  border: 1px solid #ccc;
  border-radius: 10px;
}
  
.picture {
  text-align: center;
}

.loginImage {
  width: 400px;
  border-radius: 10px;
}

.greeting {
  display: block;
  text-align: center;
}

.username, .password {
  width: 100%;
  padding: 12px 10px;
  font-size: medium;
  margin: 8px 0; 
  border: 1px solid #ccc;
  box-sizing: border-box;
}

.loginButton {
  background-color: #4085d3;
  color: white;
  font-size: medium;
  padding: 14px 10px;
  margin: 8px 0;
  border: none;
  cursor: pointer;
  width: 100%;
} 

@media screen and (min-width: 400px) {
  .loginContainer{
    width: 300px;
  }

  .loginImage{
    width: 200px;
  }
} 

@media screen and (min-width: 600px) {
  .loginContainer{
    width: 400px;
  }

  .loginImage {
    width: 250px;
  }
}

@media screen and (min-width: 1150px) {
  .loginContainer{
    width: 600px;
  }

  .loginImage {
    width: 400px;
  }
}