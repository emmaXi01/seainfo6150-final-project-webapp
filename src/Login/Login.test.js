import React from "react";
import { render } from "@testing-library/react";
import Login from "./Login.jsx";

describe("Login tests", () => {
  it("renders the login component correctly", () => {
    const { container } = render(<Login onLogin={()=>{}} />);
    expect(container).toMatchSnapshot();
  });
});