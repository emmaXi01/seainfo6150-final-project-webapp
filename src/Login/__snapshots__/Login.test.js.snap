// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login tests renders the login component correctly 1`] = `
<div>
  <div
    class="loginContainer"
  >
    <div
      class="picture"
    >
      <img
        alt="login"
        class="loginImage"
        src="login.png"
      />
    </div>
    <h4
      class="greeting"
    >
      hello :)
    </h4>
    <p
      class="error"
    />
    <div
      class="login-body"
    >
      <input
        class="username"
        placeholder="User name"
        value=""
      />
      <input
        class="password"
        placeholder="Password"
        value=""
      />
      <button
        class="loginButton"
      >
        Log In
      </button>
    </div>
  </div>
</div>
`;
